app/services/omniauth/authentication_service.rb

```diff
@@ -29,19 +29,31 @@ def policy
     end
 
     def user
-      @user ||= ::User.find_or_create_by(email: email) do |user|
-        user.email      = email
-        user.first_name = first_name
-        user.last_name  = last_name
-        user.password   = Devise.friendly_token[0, 20]
+      @user ||= identified_user || found_user
```

Comment:
I can agree that both methods\definitions should not belong here, but I was not sure what is the best place for them in your app? A model would not work since I am updating the `Identity` object, so I assume another service?

Reply:
Thanks, for your efforts, we really appreciate it! 
Yep, we have to do some clean-ups in the design of our application abstractions, so it will be refactored. Also thx for not creating new abstractions and keep it simple, it will save our time!